internal/common/platform.go

```diff
@@ -0,0 +1,24 @@
+package common
+
+import "github.com/openshift/assisted-service/models"
+
+const (
+	VmwareManufacturer string = "VMware, Inc."
+)
+
+var supportedManufacturers = map[string]models.PlatformType{
+	VmwareManufacturer: models.PlatformTypeVsphere,
+}
+
+func GetSupportedPlatforms(inventory models.Inventory) *[]models.PlatformType {
+	hostPlatform := []models.PlatformType{models.PlatformTypeBaremetal}
+	AppendPlatformFromVendor(*inventory.SystemVendor, &hostPlatform)
+	return &hostPlatform
+}
+
+func AppendPlatformFromVendor(vendor models.SystemVendor, hostsPlatform *[]models.PlatformType) {
+	platform := supportedManufacturers[vendor.Manufacturer]
+	if platform != "" {
+		*hostsPlatform = append(*hostsPlatform, platform)
```

Comment:
GetHostPlatform - try to write a immutable functions   
append in the GetSupportedPlatforms function